src/schematools/loaders.py

```diff
@@ -0,0 +1,35 @@
+import importlib
+import os
+from typing import TYPE_CHECKING, Callable, Optional
+
+from schematools.types import DatasetSchema
+from schematools.utils import dataset_schema_from_file, dataset_schema_from_url
+
+
+def get_loader(reference: str) -> Callable:
+    """Gets a loader callable from a string reference."""
+    module_name, separator, callable_name = reference.partition(":")
+    if not separator:
+        raise ValueError("Provide a fully qualified reference `package.module:callable`")
+    obj = importlib.import_module(module_name)
+    for attr in callable_name.split("."):
+        obj = getattr(obj, attr)
+    return obj
+
+
+def schema_file_loader(dataset_id: str) -> Optional[DatasetSchema]:
+    """Load a datasetschema from the filesystem."""
+    try:
+        schemas_path = os.environ["SCHEMA_PATH"]
+    except KeyError:
+        return None
+    return dataset_schema_from_file(dataset_id, schemas_path=schemas_path, prefetch_related=True)
+
+
+def schema_url_loader(dataset_id: str) -> Optional[DatasetSchema]:
+    """Load a datasetschema from the a url."""
+    try:
+        schemas_url = os.environ["SCHEMA_URL"]
+    except KeyError:
+        return None
+    return dataset_schema_from_url(schemas_url, dataset_id, prefetch_related=True)
```

Comment:
There is a discrepancy between loading schema's from URL versus files:
* `dataset_schema_from_url` calls `schematools.utils.schema_from_url`: a utility function.
* `dataset_schema_from_file` calls `DatasetSchema.from_file`: a member function defined on the `DatasetSchema` class

Why is one a member function and the other simply a utility function? Can we remedy this discrepancy in this PR?

I would argue that loading a schema, either from file or URL should not be defined in a member function on `DatasetSchema`. Hence I thing `DatasetSchema.from_file` should become an utility function as well.

Reply:
I'll look into that. This would be a breaking change, so this would cause a major version bump.

Reply:
Just a question, what's against a `DatasetSchema.from_file`? This class-method-as-factory is quite common practice in python code, or am I wrong?

Reply:
The data structure doesn't know about files, just the factory. Where you put the factory only determines how you call it, and how easy it is to discover.

Reply:
> Where you put the factory only determines how you call it

I tend to disagree. Classes are a way to encapsulate data with behavior on that data. A class representing an Amsterdam Schema dataset does not need any logic for obtaining the source of that data from the filesystem, from the network, from a bluetooth connection, or from whatever data store that comes to mind.

That kind of logic - logic for dealing with external systems - has absolutely nothing to do with the dataset data structure. As such it does not belong in the dataset class.

But as I said to @jjmurre I care for consistency more than in this specific case. Either we put the logic for loading from file system and loading from the network both on the class or  both as utility functions outside of the class. Not one on the class and the other as a utility function.

Reply:
Basically what I am saying is that methods related to I/O don't belong on the class `DatasetSchema` as `DatasetSchema` is not about I/O in any way